src/components/MapEditor/index.js

```diff
@@ -0,0 +1,68 @@
+import React, { memo , useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Map, Marker } from '@datapunt/react-maps';
+import { ViewerContainer } from '@datapunt/asc-ui';
+import { Zoom } from '@datapunt/amsterdam-react-maps/lib/components';
+import styled from '@datapunt/asc-core';
+import MAP_OPTIONS from 'shared/services/configuration/map-options';
+import BackgroundLayer from 'components/BackgroundLayer';
+import { markerIcon } from 'shared/services/configuration/map-markers';
+import { feature2location } from 'shared/services/map-location';
+
+
+const MapWrapper = styled.div`
+  position: relative;
+`;
+
+const StyledMap = styled(Map)`
+  width: 100%;
+  height: 450px;
+`;
+
+const StyledViewerContainer = styled(ViewerContainer)`
+  z-index: 400;
+`;
+
+const MapEditor = ({ location }) => {
+  const [marker, setMarker] = useState();
+
+  useEffect(() => {
+    if (!marker || !location.geometrie) return;
+
+    const opacity = 1;
+    const latlng = feature2location(location.geometrie);
+    marker.setLatLng(latlng);
+    marker.setOpacity(opacity);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [marker, location]);
+
+  return (
+    <MapWrapper>
+      <StyledMap options={MAP_OPTIONS}>
+        <StyledViewerContainer bottomRight={<Zoom />} />
+        <Marker
+          setInstance={setMarker}
+          args={[{
+            lat: 0,
+            lng: 0,
+          }]}
+          options={{
+            icon: markerIcon,
+            opacity: 0,
+          }}
+        />
```

Comment:
Shouldn't the marker only be rendered when there is an actual location available to the Map component?

Reply:
we could implement it like this, but if we do that, we won't be able to 'flyto' in the later stage :-)

Reply:
In which cases would we need `flyto`?

Reply:
extisting functionality: a marker is visible and then the user selects a location from the autosuggest.

Reply:
Still, you're rendering a marker where that is not needed. If there is a location for a marker, only then the marker should be rendered. And it can be moved to a different point on the map when another location for the marker comes in.